### PR TITLE
tests: Adds gcr.io Windows image test case

### LINKED ISSUE #74134: gcr.io Windows image needed for test
**What would you like to be added**:

/sig testing
/sig windows

In the the PR https://github.com/kubernetes/kubernetes/pull/72777 we're adding a test case for pulling a Windows image from dockerhub (``should be able to pull image from docker hub [WindowsOnly]``). It has been suggested that we should also add a test case for ``gcr.io``.

However, there are no known public Windows images on gcr.io, so the only option would be to upload it ourselves. Our proposal is to have it uploaded to the ``gcr.io/kubernetes-e2e-test-images`` or ``gcr.io/google-samples`` registries (and ``gcr.io/k8s-authenticated-test`` if we want to also cover the test case ``should be able to pull from private registry with secret`` using Windows images).

The proposed image manifest list to be uploaded is ``e2eteam/busybox:1.29`` built from [1], which is currently hosted on dockerhub [2] and is currently used for testing Kubernetes with Windows nodes [3]. The manifest list contains the images:

- e2eteam/busybox:1.29-1803 (for Windows Server 1803)
- e2eteam/busybox:1.29-1809 (for Windows Server 2019)

[1] https://github.com/kubernetes-sigs/windows-testing/blob/master/images/busybox/Dockerfile
[2] https://hub.docker.com/r/e2eteam/busybox
[3] https://testgrid.k8s.io/sig-windows#Conformance%20acs-engine%20on%20Azure

**Why is this needed**:

The gcr.io Windows images will be used for the Windows equivalent test case for the following test:

- ``should be able to pull image from gcr.io``

### LINKED ISSUE #74740: Was the new [WindowsOnly] tag approved?
There is now a `[WindowsOnly]` test tag from https://github.com/kubernetes/kubernetes/pull/72777

I remember a `[LinuxOnly]` tag being approved in SIG Architecture and SIG Testing, but I don't remember a `[WindowsOnly]` tag. 

If this was approved I think we need to broadcast it more widely as many existing test setups including  current conformance testing appear to now need to skip / exclude this tag.

cc @timothysc  @spiffxp @dims 

/area conformance
/sig architecture
/sig testing

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind failing-test

/sig windows
/sig testing

**What this PR does / why we need it**:

Adds the test "should be able to pull image from gcr.io [NodeConformance]",
which will pull the the image "gcr.io/kubernetes-e2e-test-images/windows-nanoserver:v1".
The mentioned image is a manifest list, and it works for both
Windows Server 1803 and Windows Server 2019. The manifest list
will have to be amended when a new Windows Server is released.

Adds the test "should be able to pull from private registry with secret [NodeConformance]"
which will pull the image "gcr.io/authenticated-image-pulling/windows-nanoserver:v1".
The mentioned image is a manifest list, and it works for both
Windows Server 1803 and Windows Server 2019. The manifest list
will have to be amended when a new Windows Server is released.

The command passed to the Windows Container has been changed to
"ping -t localhost", which will keep the container in the Running state,
which is required and checked by the test.

**Which issue(s) this PR fixes**:

Fixes #74134 
Fixes #74740

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
@bclau 
thanks this LGTM.

but ideally we should replace the busy box image on L209 (1.5GB) with the same `windows-nanoserver:v1` uploaded to docker hub. because this way we are only going to download ~350 MB instead of 1.85GB

i have tested this locally.

/lgtm
/priority important-soon
/assign @timothysc

**Comment by bclau** [comment]
> 
> but ideally we should replace the busy box image on L209 (1.5GB) with the same `windows-nanoserver:v1` uploaded to docker hub. because this way we are only going to download ~350 MB instead of 1.85GB

I would postpone that until the next cycle, given how close we are to the release. It is more of a wishlist item right now. Plus, as we've previously discussed, that image is based on the windows servercore image, which should already exist on the nodes, so the actual download is limited to just the busybox layers, which are a lot smaller.

**Comment by neolit123** [comment]
@bclau 

> Plus, as we've previously discussed, that image is based on the windows servercore image, which should already exist on the nodes, so the actual download is limited to just the busybox layers, which are a lot smaller.

should all images be based on nanoserver in the longterm?

**Comment by neolit123** [comment]
/hold
"[WindowsOnly]" tag has to be removed.
pinged @bclau on slack after this was opened: 
https://github.com/kubernetes/kubernetes/issues/74740

**Comment by bclau** [comment]
removed it from the new test.

**Comment by neolit123** [comment]
> removed it from the new test

has to be removed entirely:
https://github.com/kubernetes/kubernetes/blob/cb6d74ce70c731d719897e5fb7956ed8a6a98dbc/test/e2e/common/runtime.go#L360

File: test/e2e/common/runtime.go
Comment (neolit123) [inline test/e2e/common/runtime.go]: isn't this the same as `framework.NodeOSDistroIs("windows")`, which avoids passing the flag?

**Comment by bgrant0607** [comment]
cc @dims

File: test/e2e/common/runtime.go
Comment (bgrant0607) [reply test/e2e/common/runtime.go]: Given that the specified image needs to be aligned with the command, I like windowsImage as a parameter. I could also imagine an enum, for call site readability, but that would introduce the need to check for other values, so I don't really recommend it.

**Comment by neolit123** [comment]
@bclau the auth windows image is up too.
https://github.com/kubernetes/kubernetes/issues/74134#issuecomment-468471758

so you might as well add it as part of this PR.

also please remove the release note:
> Added a Windows equivalent test to ``should

`NONE` is sufficient here i'd say.

File: test/e2e/common/runtime.go
Comment (neolit123) [inline test/e2e/common/runtime.go]: aren't we supposed to test this on Windows too?

**Review by neolit123 — COMMENTED** [review]
/lgtm
/assign @timothysc

**Comment by neolit123** [comment]
needs approval at this point.

**Review by dims — APPROVED** [review]

File: test/e2e/common/runtime.go
Comment (timothysc) [inline test/e2e/common/runtime.go]: FWIW I get why this is changed to true, but busybox/windows test?  This specific test is a little weird and needs better description IMO.  I don't understand why this is windows specific, or why the logic exists above.

**Review by timothysc — COMMENTED** [review]
Need quick feedback before approve.

File: test/e2e/common/runtime.go
Comment (neolit123) [reply test/e2e/common/runtime.go]: the busybox here that is pulled will end up being Windows based (manifest list), though a comment is not present about this in the code.

@bclau might as well use an official Microsoft image here?

Comment (neolit123) [reply test/e2e/common/runtime.go]: why not push the same `v1` image we now have on GCR to the docker hub account....?

Comment (bclau) [reply test/e2e/common/runtime.go]: Indeed, the pulled image is manifest list with only Windows images. This is the Windows equivalent test of "should be able to pull image from docker hub [LinuxOnly] [NodeConformance]"

Comment (dims) [reply test/e2e/common/runtime.go]: @PatrickLang let's wait for this conversation to be resolved before we wrap up this PR

Comment (michmike) [reply test/e2e/common/runtime.go]: @neolit123  I understand that using the official msft image is better and we can moving forward but we are using busybox in many tests so far. we don't want to make all these changes now 2 days before code freeze. are you ok with that? we can file an issue and fix for 1.15

Comment (neolit123) [reply test/e2e/common/runtime.go]: @michmike i don't want to block anything.
@timothysc questioned the usage of busybox, so i proposed solutions.

also i'm not an approver for `/test` so the final call is not mine.

Comment (michmike) [reply test/e2e/common/runtime.go]: i agree with your proposal, i was just asking to postpone it a little bit :)

**Comment by timothysc** [comment]
/approve 
/lgtm cancel 
/hold  

Please add the TODO, open an issue for tracking (/assign @stevesloka) and we can call it good enough.

**Review by timothysc — APPROVED** [review]
/lgtm 
/hold cancel

**Review by michmike — APPROVED** [review]

**Pull request merged**
